app/controllers/songs_controller.rb

```diff
@@ -16,10 +16,10 @@ def index
         songs = songs.search_by_keywords(search_value) if search_value.present?
         songs = songs.where(key: params[:key]) if params[:key].present?
         songs = songs.where(tempo: params[:tempo]) if params[:tempo].present?
+        # get number of matching songs post filtering
+        records_filtered = songs.count
         songs = songs.select('id, artist, tempo, key, name, chord_sheet, spotify_uri')
 
-        # store total number of songs after filtering
-        records_filtered = songs.length
```

Comment:
`songs.length` generated this SQL query; it essentially downloaded all the matching rows at that point into an array and then calculated the size of the array. On an initial page load, there are no filters applied so it essentially downloaded the entire songs table.
```
 Song Load (6.1ms)  SELECT id, artist, tempo, key, name, chord_sheet, spotify_uri FROM "songs" /*action='index',application='GraceTunes',controller='songs'*/
  ↳ app/controllers/songs_controller.rb:22:in `block (2 levels) in index'
```